index.bs

```diff
@@ -405,38 +400,66 @@ spec: webidl
     };
   </pre>
   <p>
-    The steps to <dfn>retrieve a permission storage entry</dfn> of a
-    <a>permission storage identifier</a> are as follows:
+    The steps to <dfn export>retrieve the permission storage</dfn> of a
+    {{PermissionName}} <var>name</var> are as follows:
   </p>
   <ol>
-    <li>If the <a>user agent</a> has a {{PermissionStorage}} associated
-    with the <a>permission storage identifier</a> in its permission store,
-    it MUST return the {{PermissionStorage}}.
+    <li>Let <var>settings</var> be the <a>current settings object</a>.</li>
+    <li>
+      Let <var>store</var> be some <a>permission store</a> associated with the
+      <a>same origin</a> as <var>settings</var>. This store SHOULD be chosen
+      consistently for a given <a>environment settings object</a>.
```

Comment:
This wouldn't necessitate a per-realm store, just that every real has a single store that is authoritative for that realm.

Or is that really the case?  Or is it that there is a hierarchy of stores for a given realm and that looking for a permission requires checking each in turn?  If the realm has a store created by storing permissions for permission "A" and it queries for permission "B", which is not yet stored in the realm.  It could have a permission for "B" stored in a per-origin store, though.  This all suggests that you either have a chain of stores to check, or that you have a store for a realm that might be initially populated with information from other stores.